packages/ember-application/tests/system/dependency_injection/to_string_test.js

```diff
@@ -1,62 +1,63 @@
 import { guidFor } from 'ember-utils';
-import { ENV, context } from 'ember-environment'; // lookup, etc
-import { run } from 'ember-metal';
-import Application from '../../../system/application';
+import { context } from 'ember-environment'; // lookup, etc
 import { Object as EmberObject } from 'ember-runtime';
-import DefaultResolver from '../../../system/resolver';
+
+import { moduleFor, AutobootApplicationTestCase, ModuleBasedTestResolver } from 'internal-test-helpers';
 
 let originalLookup, App;
 
-QUnit.module('Ember.Application Dependency Injection – toString', {
-  setup() {
+moduleFor('Ember.Application Dependency Injection - toString', class extends AutobootApplicationTestCase {
+  constructor() {
+    super();
     originalLookup = context.lookup;
 
-    run(() => {
-      App = Application.create();
-      context.lookup = {
-        App: App
-      };
+    this.runTask(() => {
+      App =  this.createApplication();
     });
-
+    context.lookup = {
+      App: App
+    }
     App.Post = EmberObject.extend();
-  },
+    this.add('model:post', App.Post);
+  }
 
   teardown() {
+    super.teardown();
     context.lookup = originalLookup;
-    run(App, 'destroy');
   }
-});
-
-QUnit.test('factories', function() {
-  let PostFactory;
-  PostFactory = App.__container__.factoryFor('model:post').class;
-  equal(PostFactory.toString(), 'App.Post', 'expecting the model to be post');
-});
-
-QUnit.test('instances', function() {
-  let post = App.__container__.lookup('model:post');
-  let guid = guidFor(post);
-
-  equal(post.toString(), '<App.Post:' + guid + '>', 'expecting the model to be post');
-});
-
-QUnit.test('with a custom resolver', function() {
-  run(App, 'destroy');
-
-  run(() => {
-    App = Application.create({
-      Resolver: DefaultResolver.extend({
-        makeToString(factory, fullName) {
-          return fullName;
-        }
-      })
-    });
-  });
 
-  App.register('model:peter', EmberObject.extend());
+  ['@test factories'](assert) {
+    let PostFactory = App.__container__.factoryFor('model:post').class;
+    assert.equal(PostFactory.toString(),
+      'App.Post',
+      'expecting the model to be post');
+  }
+
+  ['@test instances'](assert) {
+    let post = App.__container__.lookup('model:post');
+    let guid = guidFor(post);
 
-  let peter = App.__container__.lookup('model:peter');
-  let guid = guidFor(peter);
+    assert.equal(post.toString(), '<App.Post:' + guid + '>', 'expecting the model to be post');
+  }
+
+  ['@test with a custom resolver'](assert) {
```

Comment:
This test does not test the global resolver, so it should have its own module. Something like:

```js
moduleFor('Ember.Application Dependency Injection - Resolver#toString', class extends ApplicationTestCase {
  constructor() {
    super();
    return this.visit('/');
  }
  get applicationOptions {
    return assign(super.applicationOptions, {
      Resolver: class extends ModuleBasedTestResolver {
        makeToString(factory, fullName) {
          return fullName;
        }
      }
    });
  }

  ['@test toString called on a resolver'](assert) {
    this.add('model:peter', EmberObject.extend());
    
    let peter = this.applicationInstance.lookup('model:peter');
    let guid = guidFor(peter);
    assert.equal(
      peter.toString(), `<model:peter:${guid}>`,
      'expecting the supermodel to be peter'
    );
  }
```